redbot/core/core_commands.py

```diff
@@ -443,6 +482,495 @@ async def uptime(self, ctx: commands.Context):
+    @commands.cooldown(1, 1800, commands.BucketType.user)
+    @mydata.command(cls=commands.commands._AlwaysAvailableCommand, name="3rdparty")
+    async def mydata_3rd_party(self, ctx: commands.Context):
+        """ View the End User Data statements of each 3rd-party module. """
+
+        # Can't check this as a command check, and want to prompt DMs as an option.
+        if not ctx.channel.permissions_for(ctx.me).attach_files:
+            ctx.command.reset_cooldown(ctx)
+            return await ctx.send("I need to be able to attach files (try in DMs?)")
+
+        statements = {
+            ext_name: getattr(ext, "__red_end_user_data_statement__", None)
+            for ext_name, ext in ctx.bot.extensions.items()
+            if not ext.__package__.startswith("redbot.")
+        }
+
+        if not statements:
+            return await ctx.send(
+                "This instance does not appear to have any 3rd party extensions loaded."
+            )
+
+        parts = []
+
+        formatted_statements = []
+
+        no_statements = []
+
+        for ext_name, statement in sorted(statements.items()):
+            if not statement:
+                no_statements.append(ext_name)
+            else:
+                formatted_statements.append(
+                    f"### {entity_transformer(ext_name)}" f"\n\n{entity_transformer(statement)}"
+                )
+
+        if formatted_statements:
+            parts.append(
+                "## 3rd party End User Data statements"
+                "\n\nThe following are statements provided by 3rd-party extensions."
+            )
+            parts.extend(formatted_statements)
+
+        if no_statements:
+            parts.append("## 3rd-party extensions without statements\n")
+            for ext in no_statements:
+                parts.append(f"\n - {entity_transformer(ext)}")
+
+        generated = markdown.markdown("\n".join(parts), output_format="html")
+
+        html = "\n".join((PRETTY_HTML_HEAD, generated, HTML_CLOSING))
+
+        fp = io.BytesIO(html.encode())
+
+        await ctx.send(file=discord.File(fp, filename="3rd-party.html"))
+
+    async def get_serious_confirmation(self, ctx: commands.Context, prompt: str) -> bool:
+
+        confirm_token = "".join(random.choices((*ascii_letters, *digits), k=8,))
+
+        await ctx.send(f"{prompt}\n\n{confirm_token}")
+        try:
+            message = await ctx.bot.wait_for(
+                "message",
+                check=lambda m: m.channel.id == ctx.channel.id and m.author.id == ctx.author.id,
+                timeout=30,
+            )
+        except asyncio.TimeoutError:
+            await ctx.send(_("Did not get confirmation, cancelling."))
+        else:
+            if message.content.strip() == confirm_token:
+                return True
+            else:
+                await ctx.send(_("Did not get a matching confimration, cancelling."))
```

Comment:
Code suggestion for what Flame said [here](https://github.com/Cog-Creators/Red-DiscordBot/pull/4045/files#r460463811):
```suggestion
                await ctx.send(_("Did not get a matching confirmation, cancelling."))
        return False
```

Reply:
The comment link I added was actually pointing to the `return False` which hasn't been added here yet.

Reply:
Handled, this was functionally not an issue, but the type-hint is correct now.